### PR TITLE
aaaa: Ensure AAAA answers are lowercase

### LINKED ISSUE #113: dns_aaaa should return lowercase characters instead
ipv6 must be represented as lowercase characters, see https://datatracker.ietf.org/doc/html/rfc5952#section-4.3

### PR DESCRIPTION
**Pull request by b4ldr**
As per https://datatracker.ietf.org/doc/html/rfc5952#section-4.3 ensure ipv6 addresses are lowercase
fixes #113

### CODE REVIEW

File: lib/puppet/functions/dns_aaaa.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def dns_aaaa(record)
     Resolv::DNS.new.getresources(
       record, Resolv::DNS::Resource::IN::AAAA
     ).map do |res|
-      res.address.to_s
+      res.address.to_s.downcase
```
Comment (ekohl) [inline lib/puppet/functions/dns_aaaa.rb:22]: I don't know what it returns, but you can consider using `IPAddr.new(res.address.to_s).to_s` to normalize it.

Comment (b4ldr) [reply lib/puppet/functions/dns_aaaa.rb:22]: updated, although that does come with a different [5953 violation](https://github.com/ruby/resolv/pull/25) :D, but either way its the better way to go

Comment (bastelfreak) [reply lib/puppet/functions/dns_aaaa.rb:22]: Google does case randomization in queries, is it a good idea to force everything to lowercase?
https://groups.google.com/g/public-dns-discuss/c/aHSyiIlBfjo

Comment (b4ldr) [reply lib/puppet/functions/dns_aaaa.rb]: This type of randomisation aka 0x20 is only important for the answer the cache receives, this code asks the cache and then just normalises the output so not an issue for this use case

I would also be curious if google randomise the case of aaaa responses, i suspect they would honer rfc5953 for that case but didn't check

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**